br/pkg/lightning/restore/restore.go

```diff
@@ -955,8 +944,6 @@ func (rc *Controller) saveStatusCheckpoint(ctx context.Context, tableName string
 
 // listenCheckpointUpdates will combine several checkpoints together to reduce database load.
 func (rc *Controller) listenCheckpointUpdates() {
-	defer rc.checkpointsWg.Done()
```

Comment:
why move this defer?

Reply:
failpoint will call `checkpointWg.Done` manually. This is for avoid panic like `panic: sync: negative WaitGroup counter`.

Reply:
maybe add a comment, it violate normal convention to wait goroutine.

Reply:
Added.